src/core/ReactUpdateQueue.js

```diff
@@ -40,20 +41,27 @@ function getInternalInstanceReadyForUpdate(publicInstance, callerName) {
   );
 
   var internalInstance = ReactInstanceMap.get(publicInstance);
-  invariant(
-    internalInstance,
-    '%s(...): Can only update a mounted or mounting component. ' +
-    'This usually means you called %s() on an unmounted ' +
-    'component.',
-    callerName,
-    callerName
-  );
-  invariant(
-    internalInstance !== ReactLifeCycle.currentlyUnmountingInstance,
-    '%s(...): Cannot call %s() on an unmounting component.',
-    callerName,
-    callerName
-  );
```

Comment:
You reviewed it :wink:

I remember us saying that it also wasn't really harmful to call setState after unmounting. It might be indicative of a leak but that's what the warning is for for. I think the biggest argument is Promises. `fetch(url).then((result) => {this.setState({data: result})})`

If we don't want people having to check isMounted first, then we needed to make this safe.